pkgs/build-support/rust/import-cargo-lock.nix

```diff
@@ -199,12 +199,17 @@ let
         # Cargo is happy with empty metadata.
         printf '{"files":{},"package":null}' > "$out/.cargo-checksum.json"
 
-        # Set up configuration for the vendor directory.
+        # Set up configuration for the vendor directory with package name and version.
         cat > $out/.cargo-config <<EOF
+        #pkg: ${pkg.name}
+        #version: ${pkg.version}
         [source."${gitParts.url}${lib.optionalString (gitParts ? type) "?${gitParts.type}=${gitParts.value}"}"]
         git = "${gitParts.url}"
         ${lib.optionalString (gitParts ? type) "${gitParts.type} = \"${gitParts.value}\""}
-        replace-with = "vendored-sources"
+        replace-with = "vendored-sources-git-${gitParts.sha}"
+        [source.vendored-sources-git-${gitParts.sha}]
+        directory = "cargo-vendor-dir/git-${gitParts.sha}"
+        
```

Comment:
Checking EditorConfig fails, but I would like to include a blank line for readability.

Reply:
editor config doesn't complain about blank line, it complains about training whitespace on this line